### ISSUE TITLE
Improve Blankenbach convergence

### ISSUE THREAD

**Issue by tjhei** [geodynamics/aspect#1389]
Even when starting with the final, steady-state temperature, it takes 100-1000 timesteps to see convergence. Details are discussed in #1378.

I assume this is because the entropy viscosity in timestep 1 is computed differently and causes a large deviation from the initial condition.

Is there a bug? I am unclear why we use a different formula for the entropy for the first two timesteps even though it should only be needed for timestep 0.

Otherwise, can we create an additional option that basically starts at timestep 2 and uses the initial T for timestep 0 and 1? That shouldn't be difficult to get right.

**Comment by bangerth** [comment]
Yes, it's weird that you have a converged solution, and when you start with it it takes another few hundred iterations to get back to it. Are you sure it's because of the entropy viscosity? We presumably don't have the option to read in both the current and a previous solution and declare that we are already starting with the BDF2 time stepper. But I suspect that if you hacked something like that into the code, that it would help you find where exactly things go off the rails.

There is, of course, the option to just write a checkpoint, and start from it but with a different input file. If I remember correctly, there is even a test that ensures that that works correctly.

**Comment by tjhei** [comment]
> Are you sure it's because of the entropy viscosity?

Yes, pretty sure. Timestep 1 has the correct (first five digits) statistics, but timestep 2 has about 1% error and the only big difference is the entropy viscosity that is 2x bigger.

> We presumably don't have the option to read in both the current and a previous solution and declare that we are already starting with the BDF2 time stepper. But I suspect that if you hacked something like that into the code, that it would help you find where exactly things go off the rails.

Yeah, maybe I should try that next.

> There is, of course, the option to just write a checkpoint, and start from it but with a different input file. If I remember correctly, there is even a test that ensures that that works correctly.

Certainly, but this is not portable and I kind of want to ship with these files to allow quick computations of the benchmark.

**Comment by tjhei** [comment]
> Are you sure it's because of the entropy viscosity?

If I turn entropy viscosity off (beta=cr=0), I am reaching a steady state more or less immediately (it takes the minimum amount of the steady state detection termination criterium to kick in). The solutions are still mesh dependent though (using the 32x32 solution for 16x16 increases the number of iterations slightly).

> But I suspect that if you hacked something like that into the code, that it would help you find where exactly things go off the rails.

This also works: if I remove the special case for ``timestep_number <=1`` and use the correct velocity in the entropy viscosity computation, I also converge very quickly.
